wrapanapi/rhevm.py

```diff
@@ -356,12 +356,37 @@ def _get_vm_nic_service(self, vm_name, nic_name):
     def _get_vm_nic(self, vm_name, nic_name):
         return self._get_vm_nic_service(vm_name, nic_name).get()
 
-    def update_vm_nic(self, vm_name, network_name, nic_name='nic1',
-                      interface=types.NicInterface.VIRTIO):
+    def _get_network(self, network_name):
+        """retreive a network object by name"""
+        networks = self.api.system_service().networks_service().list(
+            search='name={}'.format(network_name))
+        try:
+            return networks[0]
+        except IndexError:
+            raise ItemNotFound('No match for network by "name={}"'.format(network_name))
+
+    def update_vm_nic(self, vm_name, network_name, nic_name='nic1', interface='VIRTIO'):
         nic = self._get_vm_nic(vm_name, nic_name)
-        nic_service = self._get_vm_nic_service(vm_name, nic_name)
-        nic.network = types.Network(name=network_name)
-        nic.interface = interface
+        nic.network = self._get_network(network_name)
+        vnic_profiles = [v
+                        for v in self.api.system_service().vnic_profiles_service().list()
```

Comment:
I think these .list() APIs might have a filter kwarg that would return the filtered list for us already?

Reply:
some have a `search` which takes a `'<key>=<name>'` string, but these ones did not appear to, only a query parameter.

Reply:
Yeah I agree, I looked around and give up as well ;)